code/game/objects/weapons.dm

```diff
@@ -2,6 +2,32 @@
 	name = "weapon"
 	icon = 'icons/obj/weapons.dmi'
 
+	var/melee_rename = 0
+	var/melee_reskin = 0
+	var/mreskinned = 0
+	var/list/moptions = list()
+
+/obj/item/weapon/proc/rename_wopit(mob/M)
+	var/input = stripped_input(M,"What do you want to name the [name]?", ,"", MAX_NAME_LEN)
+
+	if(src && input && !M.stat && in_range(M,src) && !M.restrained() && M.canmove)
+		name = input
+		M << "You name the [name] [input]. Say hello to your new friend."
+		return
+
+/obj/item/weapon/proc/reskin_wopit(mob/M)
+	var/choice = input(M,"Warning, you can only reskin your [name] once!","Reskin [name]") in moptions
+
+	if(src && choice && !M.stat && in_range(M,src) && !M.restrained() && M.canmove)
+		if(moptions[choice] == null)
+			return
+		else
+			icon_state = moptions[choice]
+			item_state = moptions[choice]
```

Comment:
Around here is where you'd want to reset your weapon sounds. I'd just organize the soundbits to be named after their moptions name, and do something like "hitsound = '/sound/weapons/[moptions[choice]].ogg'" Not sure if you can use that syntax specifically for that function, but I'm fairly certain its possible. Or doing a length if statement, but that isn't wanted I assume.